src/api/libcellml/model.h

```diff
@@ -256,18 +256,19 @@ class LIBCELLML_EXPORT Model: public ComponentEntity
      *
      * Traverses the model looking for @c Units attached to
      * @c Variables that are not standard units and which are not
-     * linked to @c Units added to the model.
+     * linked to @c Units added to the model.  This method will link
+     * variable units specified by name to units in the model
+     * (if they are found). Any variable units that cannot be linked
+     * to units in the model are left untouched.
      *
      * Unlinked variable units can occur when a @c Variable's units are
-     * set by name.  If a @c Units cannot be found in the model that matches
-     * the name given to the variables setUnits() method then the units for
-     * that variable will be unlinked.  This method will link variable units
-     * specified by name to units in the model (if they are found). Any variable
-     * units that cannot be linked to units in the model are left untouched.
-     *
-     * Any @c Variables found that are not linked to model @c Units will be
-     * linked. If a Variable has units that are not attached to the model
-     * then the units will be added to the model as well.
+     * set by name.  If the model that the variable belongs to has
+     * units defined with the same name then that variables units will
```

Comment:
Minor grammar/formatting stuff:
"If the @c Model to which the @c Variable belongs has @c Units defined with the same name, then that @c Variable's @c Units will not be linked to the @c Model's @c Units."